allennlp/models/semantic_parsing/wikitables/wikitables_decoder_state.py

```diff
@@ -86,12 +91,21 @@ def __init__(self,
         self.output_action_embeddings = output_action_embeddings
         self.action_biases = action_biases
         self.action_indices = action_indices
+        # This is just the reverse of action indices.
+        self.global_to_batch_action_indices: Dict[Tuple[int, int], int] = {}
+        for (batch_index, batch_action_index), global_index in action_indices.items():
+            if global_index == -1:
+                # The batch_action_index here corresponds to a linked action. We don't need to map
+                # it.
+                continue
+            self.global_to_batch_action_indices[(batch_index, global_index)] = batch_action_index
         self.possible_actions = possible_actions
         self.flattened_linking_scores = flattened_linking_scores
         self.actions_to_entities = actions_to_entities
         self.entity_types = entity_types
         self.world = world
         self.example_lisp_string = example_lisp_string
+        self.checklist_state = checklist_state
```

Comment:
I can understand wanting to still have `None` as the default value to this constructor, so that, e.g., the MML parser doesn't have to pass in a list of `Nones`, which would be annoying.  But you could convert the `None` to a `[None, ...]` list inside the constructor here, and that would let you simplify the logic everywhere else.

Reply:
Done.